File: packages/python/src/alumnium/tools/click_tool.py
from pydantic import Field

from ..drivers.base_driver import BaseDriver
from .base_tool import BaseTool
from .upload_tool import UploadTool


class ClickTool(BaseTool):
    __doc__ = (
        "Click an element."
        " NEVER use ClickTool on comboboxes, instead proceed to click the desired option directly."
        f" NEVER use ClickTool to upload files - use {UploadTool.__name__} instead."
    )

    id: int = Field(description="Element identifier (ID)")

    def invoke(self, driver: BaseDriver):
        driver.click(self.id)

File: packages/java/lib/src/main/java/ai/alumnium/tool/ClickTool.java
package ai.alumnium.tool;

import ai.alumnium.driver.BaseDriver;
import ai.alumnium.tool.annotation.ToolDescription;
import ai.alumnium.tool.annotation.ToolField;

@ToolDescription("Click an element. NEVER use ClickTool to upload files - use UploadTool instead.")
public record ClickTool(@ToolField(description = "Element identifier (ID)") int id) implements BaseTool {
    @Override
    public void invoke(BaseDriver driver) {
        driver.click(id);
    }
}